### PR TITLE
overlord/ifacestate: add support for Disconnect

### PR DESCRIPTION
**Pull request by zyga**
This branch changes stub Disconnect() to really disconnect the plug from
the slot. The code layout mimics the earlier work on Connect().

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

**Comment by zyga** [comment]
retest this please

**Comment by pedronis** [comment]
+1

File: overlord/ifacestate/ifacemgr.go [hunk 1]
```diff
@@ -61,7 +61,14 @@ func Connect(change *state.Change, plugSnap, plugName, slotSnap, slotName string
 }
 
 // Disconnect initiates a change disconnecting an interface.
-func (m *InterfaceManager) Disconnect(plugSnap, plugName, slotSnap, slotName string) error {
+func Disconnect(change *state.Change, plugSnap, plugName, slotSnap, slotName string) error {
+	// TODO: Remove the intent-to-connect from the state so that we no longer
+	// automatically try to reconnect on reboot.
+	summary := fmt.Sprintf(i18n.G("Disconnecting %s:%s from %s:%s"),
```
Comment (niemeyer) [inline overlord/ifacestate/ifacemgr.go]: This is a task.. "Disconnect" rather than "Disconnecting"

Comment (niemeyer) [inline overlord/ifacestate/ifacemgr.go]: "Connecting" should be fixed accordingly above

**Comment by niemeyer** [comment]
LGTM

**Pull request merged**